### PR TITLE
sidewalks to have elevation and curb

### PR DESCRIPTION
**Pull request by srothst1**

### CODE REVIEW

**Comment by srothst1** [comment]
Screen Shot 2022-06-24 at 3 25 55 PM

**Comment by srothst1** [comment]
Screen Shot 2022-06-27 at 8 13 05 PM

Two issues:
- The x axis spacing of the curbs is not correct. Input here would be helpful.
- The z axis is incorrectly aligned with the sidewalk. This should be easier to resolve.

File: src/aframe-streetmix-parsers.js [hunk 1]
```diff
@@ -524,9 +524,28 @@ function createSegmentElement (scaleX, positionX, positionY, rotationY, mixinId,
...
-
   segmentEl.setAttribute('rotation', '270 ' + rotationY + ' 0');
   segmentEl.setAttribute('mixin', mixinId);
+
+  //TODO: create a new mixin object that acts as a curb
+  if (mixinId == 'sidewalk'){
+    var leftCurb = document.createElement('a-entity');
+    leftCurb.setAttribute('id', "left curb");
+    //-(scaleX*3*3.28084)*0.5
+    leftCurb.setAttribute('position', 'x', -(scaleX*3*0.5));
```
Comment (kfarr) [inline src/aframe-streetmix-parsers.js:535]: `    leftCurb.setAttribute('position', 'x', -(segmentWidthInMeters / 2)); `

`    rightCurb.setAttribute('position', 'x', (segmentWidthInMeters / 2)); `

The `segmentWidthInMeters` variable for the current segment will need to be passed to this function

Comment (srothst1) [reply src/aframe-streetmix-parsers.js:535]: Hi @kfarr. I tried this and it did not resolve the issue.

**Comment by kfarr** [comment]
Work in progress screenshot: boxes instead of planes for sidewalk with random color instead of texture
image

**Comment by kfarr** [comment]
Also regarding vertical dimensions of the sidewalk "concrete" box:
- it is 0.4m around 0; this means it is 0.2m (20 cm) above the current street planes and 20cm above the previous sidewalk elevation. This also means the block starts flush with the -0.2 placement of ground planes.
- This will require vertical (y) adjustment by +0.2m of pedestrian and other sidewalk prop objects

image

**Comment by kfarr** [comment]
other things to do:
- [ ] - change texture
- [ ] - test for different street lengths
- [ ] - test for other sidewalk or sidewalk-like segment types (bike parking, bus stop)
- [ ] - change street props y positions +0.2
- [ ] - change pedestrians y positions +0.2
- [ ] - consider impact on building height or other "ground" types such as parking lot, seawall
- [ ] - create separate ticket for updating intersection component to also use box geometry

**Comment by kfarr** [comment]
this looks good to me, I suggest to merge this

**Pull request merged**